### PR TITLE
feat; Catalog caching & lineage information 

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
#minor Add support for providing Catalog information for enabled node executions. Also added support for cases when Cache was populated / hit, miss or disabled

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
Before this change, Flyte Admin had no concept of Catalog Information and Catalog lookup identifiers

## Tracking Issue
https://github.com/lyft/flyte/issues/138

## Follow-up issue
Once this is ready, UI can easily add information to the UI to show the various states of Catalog
https://github.com/lyft/flyte/issues/71

### CODE REVIEW

File: pkg/repositories/models/node_execution.go [hunk 1]
```diff
@@ -36,5 +36,7 @@ type NodeExecution struct {
 	// Execution Error Kind. nullable, can be one of core.ExecutionError_ErrorKind
 	ErrorKind *string `gorm:"index"`
 	// Execution Error Code nullable. string value, but finite set determined by the execution engine and plugins
-	ErrorCode *string
+	ErrorCode   *string
+	// TODO, This is never really null after this point
```
[the remark this answers is not in the document]
Comment (kumare3) [reply pkg/repositories/models/node_execution.go:40]: i will remove the comment, its wrong
